src/Persistence_matrix/include/gudhi/Fields/Multi_field.h

```diff
@@ -0,0 +1,486 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber, Clément Maria
+ *
+ *    Copyright (C) 2022-24 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+/**
+ * @file Multi_field.h
+ * @author Hannah Schreiber, Clément Maria
+ * @brief Contains the @ref Multi_field_element class.
+ */
+
+#ifndef MATRIX_FIELD_MULTI_H_
+#define MATRIX_FIELD_MULTI_H_
+
+#include <utility>
+#include <vector>
+#include <gmpxx.h>
+#include <stdexcept>
+
+namespace Gudhi {
+namespace persistence_fields {
+
+/**
+ * @class Multi_field_element Multi_field.h gudhi/Fields/Multi_field.h
+ * @ingroup persistence_fields
+ *
+ * @brief Class representing an element of a multi-field.
+ * The characteristics will corresponds to all prime numbers in the interval given as template.
+ *
+ * @tparam minimum Interval closed lower bound.
+ * @tparam maximum Interval closed upper bound.
+ */
+template <unsigned int minimum, unsigned int maximum>
+class Multi_field_element {
+ public:
+  using element_type = mpz_class;           /**< Type for the elements in the field. */
+  using characteristic_type = element_type; /**< Type for the field characteristic. */
+
+  /**
+   * @brief Default constructor. Sets the element to 0.
+   */
+  Multi_field_element();
+  /**
+   * @brief Constructor setting the element to the given value.
+   *
+   * @param element Value of the element.
+   */
+  Multi_field_element(element_type element);
+  /**
+   * @brief Copy constructor.
+   *
+   * @param toCopy Element to copy.
+   */
+  Multi_field_element(const Multi_field_element& toCopy);
+  /**
+   * @brief Move constructor.
+   *
+   * @param toMove Element to move.
+   */
+  Multi_field_element(Multi_field_element&& toMove) noexcept;
+
+  /**
+   * @brief operator+=
+   */
+  friend void operator+=(Multi_field_element& f1, Multi_field_element const& f2) {
+    mpz_add(f1.element_.get_mpz_t(), f1.element_.get_mpz_t(), f2.element_.get_mpz_t());
```

Comment:
```suggestion
    f1.element_ += f2.element_;
```
?
(for mpz_mod, you indeed have to go for the long form (at least for subtraction), because `%` computes something slightly different for negative numbers)